### DISCUSSION TITLE
SNMP TV3Auth & TV3Priv

### DISCUSSION THREAD

**Discussion by NgaKamleiter** [geby/synapse#10] [General]
Hello everyone,
I am Michael.

I develop software for a medium-sized company in Germany.
I am currently building software that can query information via various protocols.

One of the protocols is SNMP.
This is where I came across the synapse library, which is really great.
Well done to everyone!

What I noticed when I wanted to use SNMPV3 was the outdated hash algorithm.
Unfortunately, only SHA1 is available for selection, which is no longer used by most SNMP agents.

Even with the selectable encryption standard AES, I am not sure whether the current standard AES256 is already used here.

Are there any plans to update the hash algorithm and the encryption standard in the future?

I would be really pleased as I would like to integrate the synapse library into my software.

Best regards
Michael

**Comment by geby** [comment]
Hi,

Synapse need various Crypto functions (for cryptography and hashing) on many places. So, some basic hashes are built-in, however it is little outdated now. And I am not able to create lot of new algorhytms. Better will be to use some external crypto libraries, what is updated, optimized and safe. I have to think about it!

What about some flexible plugin model, like is for SSL currently? (Or as part of SSL plugin directly?)

Lukas

**Reply by NgaKamleiter** [reply]
Hello Lukas,
Thank you for your reply.

Yes, I also think that external crypto libraries would be a good solution here. The requirements and standards are changing really quickly these days. I can fully understand that it is not possible to keep the crypto functions in Synapse up to date.

A plugin model would be a very good solution!

Best regards
Michael
